tests/10apidoc/03events-initial.pl

```diff
@@ -126,9 +127,13 @@ sub flush_events_for
    });
 }
 
+# Note that semantics are undefined if calls are interleaved with differing
+# $room_ids for the same user.
+#
+# if $room_id is undefined, all joined rooms will be listened to.
 sub await_event_for
 {
-   my ( $user, $filter ) = @_;
+   my ( $user, $filter, $room_id ) = @_;
```

Comment:
Actually thinking about it, that's not a lot neater. Maybe nicer to say all named arguments, but if there's only one value presume for back-compat that it's the filter sub.

    my $user = shift;
    my %params = @_ == 1 ? { filter => $_[0] } : @_;

Then change the later `$filter` to `$params{filter} || sub {1}`